### PR TITLE
refactor: bazel

### PR DESCRIPTION
**Pull request by cgrindel**
- Refactor `bazel.bazel`
  - Remove `workspace.Finder`.
  - Add `workspaceRoot` to store the path to the root of the Bazel workspace.
- Update `bazel.New()` to take a `workspaceRoot` parameter.
- Add `bazel.Find()` and `bazel.FindFromWd()` to find a workspace root and return an initialized `Bazel` instance.
- Refactor `Bazel` interface
  - Remove `WithOverrideWorkspaceRoot()`. Instances can now be created using a path.
  - Remove `Spawn()`. Appears to be an alias for `RunCommand()`.
  - Update `WithEnv()` to not create a new instance when applying the environment variables.

### CODE REVIEW

File: pkg/bazel/bazel.go [hunk 1]
```diff
@@ -28,7 +28,6 @@ import (
 	"aspect.build/cli/bazel/flags"
 	"aspect.build/cli/pkg/bazel/workspace"
 	"aspect.build/cli/pkg/ioutils"
-
```
Comment (f0rmiga) [inline pkg/bazel/bazel.go:31]: Unnecessary change. We actually like that first-party imports are a separate block from third-party block.

File: pkg/bazel/bazel.go [hunk 2]
```diff
@@ -110,29 +114,13 @@ func (*bazel) createRepositories() *core.Repositories {
 	return core.CreateRepositories(gcs, gcs, gitHub, gcs, gcs, true)
 }
 
-// Spawn is similar to the main() function of bazelisk
-// see https://github.com/bazelbuild/bazelisk/blob/7c3d9d5/bazelisk.go
-func (b *bazel) Spawn(command []string, streams ioutils.Streams) (int, error) {
```
Comment (f0rmiga) [inline pkg/bazel/bazel.go:115]: Interesting that we had this. :)
Thanks for cleaning up.

File: pkg/bazel/workspace/not_found_error.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+package workspace
+
+import (
+	"fmt"
+)
+
+type NotFoundError struct {
```
Comment (f0rmiga) [inline pkg/bazel/workspace/not_found_error.go:23]: Would you mind adding godoc comments to these?

**Review by f0rmiga — APPROVED** [review]
It's a good improvement.

File: pkg/bazel/bazel.go (hunk 1 above)
Comment (cgrindel) [reply pkg/bazel/bazel.go:31]: Reverted.

File: pkg/bazel/workspace/not_found_error.go (hunk 3 above)
Comment (cgrindel) [reply pkg/bazel/workspace/not_found_error.go:24]: Done.
